linker/shared/src/main/scala/org/scalajs/linker/backend/javascript/SourceMapWriter.scala

```diff
@@ -171,52 +193,174 @@ object SourceMapWriter {
 
     private def writePendingSegment(): Unit = {
       if (pendingColumnInGenerated >= 0) {
-        // Allocate a name string *before* we write a fragment so we can cache it.
-        val originalName =
-          if (pendingName.isDefined) pendingName.get.toString()
-          else null
+        if (pendingPos.isEmpty) {
+          doWriteSegment(pendingColumnInGenerated, null, 0, 0, null)
+        } else {
+          // Allocate a name string *before* we write a fragment so we can cache it.
+          val originalName =
+            if (pendingName.isDefined) pendingName.get.toString()
+            else null
+
+          doWriteSegment(pendingColumnInGenerated,
+              pendingPos.source, pendingPos.line, pendingPos.column, originalName)
+        }
+      }
+    }
+
+    private def readRawVLQ(buf: ByteBuffer): Int = {
+      var shift = 0
+      var value = 0
 
-        doWriteSegment(pendingColumnInGenerated, pendingPos, originalName)
+      while ({
+        val i = buf.get()
+        value |= (i & 127) << shift
+        (i & 128) != 0
+      }) {
+        shift += 7
       }
+
+      val neg = (value & 1) != 0
+      value >>>= 1
+
+      if (!neg) value
+      else if (value == 0) Int.MinValue
```

Comment:
Given what the numbers we read represent, `Int.MinValue` is not a realistic input for `writeRawVLQ`. Consider getting rid of this case to avoid one branch. (It might even remove *two* branches at the assembly level since it will basically become a `SELECT`.)